### DISCUSSION TITLE
Dynamic constraints combined with 'soft' constraining

### DISCUSSION THREAD

**Discussion by filipdoai** [fvutils/pyvsc#162] [Q&A]
So question is, if it is possible to have a 'dynamic' constraint block which contains 'soft' constraints.
Idea is to have generally random items with possibility to have a dynamic constraint which fixes all fields, but still allows some of them to be changed in a 'randomize_with' block.

```
@vsc.randobj
class RandA:
    def __init__(self):
        self.a_field = vsc.rand_uint8_t(8)
        self.b_field = vsc.rand_uint8_t(8)

    @vsc.constraint
    def default_c(self):
        self.a_field < 100
        self.b_field < 200

    
    @vsc.dynamic_constraint
    def fixed_c(self):
        vsc.soft(self.a_field == 8)
        vsc.soft(self.b_field == 8)

rand_a = RandA()

with rand_a.randomize_with(solve_fail_debug=1) as it:
    it.fixed_c()
    it.a_field == 10

print(rand_a.a_field)
```

The debug message is: 
```
Problem Set: 2 constraints
  <unknown>:
    constraint fixed_c {
(a_field == 8)(b_field == 8)}
;
  <unknown>:
    (a_field == 10);
```

**Comment by mballance** [comment]
Hi @filipdoai, this definitely should be possible. Dynamic constraints exist to support encapsulation, and encapsulating soft constraints is certainly fair game. Let me have a closer look at your example (thanks for that, BTW!)
